hw2/04-background.js

```diff
@@ -1 +1,56 @@
 // Add your code here
+document.addEventListener("DOMContentLoaded", () => {
+  const bodyElement = document.body;
+  const intervalInput = document.getElementById("intervalInput");
+  const toggleButton = document.getElementById("toggleButton");
+  let intervalId = null;
+  let isRunning = false;
+
+  // Function to change the background color of the body to a random semi-transparent color
+  const changeBackgroundColor = () => {
+    // Generate a random color using RGBA format with 50% opacity
+    const color = `rgba(${Math.floor(Math.random() * 256)}, ${Math.floor(
+      Math.random() * 256
+    )}, ${Math.floor(Math.random() * 256)}, 0.5)`;
+    bodyElement.style.backgroundColor = color;
+  };
+
+  // Function to start changing the background color at a given interval
+  const startChangingBackground = (interval) => {
+    if (intervalId) clearInterval(intervalId);
+    // Setting up a new interval that changes the background color every 'interval' seconds
+    intervalId = setInterval(changeBackgroundColor, interval * 1000);
```

Comment:
I recommend putting in at least one line of code that makes it clear you're converting the user's input from seconds to milliseconds. Otherwise it might be unclear why you're multiplying 'interval' by 1000.

Reply:
Added the comment to denote the conversion is happening in milliseconds.